File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l,r=0,len(height)-1
        max_=0
        while l<r:
            length=r-l
            h=min(height[l],height[r])
            max_=max(max_,length*h)
            if height[l]<=height[r]:
                l+=1
            else:
                r-=1
        return max_

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int i=0;
        int j=height.length-1;
        int maxA=0;
        while(i<j){
            int dif=j-i;
            int depth=Math.min(height[i],height[j]);
            maxA=Math.max(maxA,depth*dif);
            if(height[j]>height[i]){
                i++;
            }else{
                j--;
            }
        }
        return maxA;
    }
}